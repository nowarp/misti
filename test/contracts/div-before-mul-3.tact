fun test(): Int {
  let a: Int = 10 / 3; // Division operation
  let b: Int = 3;
  let c: Int = 2;
  let taintedWitha1: Int = 2 - a;
  let taintedWitha2: Int = 55 + taintedWitha1;
  let result: Int = taintedWitha2 * c; // Error: Using a variable tainted with division
  return result;
}

