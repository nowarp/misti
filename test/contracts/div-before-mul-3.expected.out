test/contracts/div-before-mul-3.tact:3:3:
  2 |   let a: Int = 10 / 3; // Division operation
> 3 |   let b: Int = 3;
        ^
  4 |   let c: Int = 2;
Variable is never used